index.ts

```diff
@@ -369,9 +363,471 @@ const CreateOrgSchema = OrgSchema.pick({
   createdBy: true,
 });
 
-// Export all schemas
+const parseCommaSeparated = (value: string | undefined): string[] => {
+  if (!value) return [];
+  return value.split(',').map(s => s.trim()).filter(s => s);
+};
+
+const normalizeCsvData = (data: Record<string, unknown>): Record<string, unknown> => {
+  const normalized: Record<string, unknown> = {};
+  Object.keys(data).forEach(key => {
+    const value = data[key];
+    const normalizedKey = key.toLowerCase();
+    normalized[normalizedKey] = value === '' || value === null ? undefined : value;
+  });
+  return normalized;
+};
+
+const calculateAge = (birthMonth: number, birthYear: number): number => {
+  const today = new Date();
+  const currentYear = today.getFullYear();
+  const currentMonth = today.getMonth() + 1;
+  let age = currentYear - birthYear;
+  if (currentMonth < birthMonth) {
+    age--;
+  }
+  return age;
+};
+
+const validateChildAge = (month: string | undefined, year: string | undefined): boolean => {
+  if (!month || !year) return true;
+  const birthMonth = parseInt(month);
+  const birthYear = parseInt(year);
+  if (isNaN(birthMonth) || isNaN(birthYear)) return true;
+  return calculateAge(birthMonth, birthYear) < 18;
+};
+
+const AddUsersCsvSchema = z.object({
+  id: z.string().trim().optional(),
+  usertype: z.preprocess(
+    (val) => {
+      if (val === undefined || val === null || val === '') return '';
+      return typeof val === 'string' ? val.trim().toLowerCase() : String(val);
+    },
+    z.string().superRefine((val, ctx) => {
+      if (!val || val.trim() === '') {
+        ctx.addIssue({
+          code: ZodIssueCode.custom,
+          message: 'userType is required'
+        });
+        return;
+      }
+      if (!['child', 'caregiver', 'teacher'].includes(val)) {
+        ctx.addIssue({
+          code: ZodIssueCode.custom,
+          message: 'userType must be one of: child, caregiver, teacher'
+        });
+      }
+    })
```

Comment:
This is more verbose than necessary and doesn't utilize Zod primitives. It could be rewritten like below. 

```js
const UserType = z.enum(["child", "caregiver", "teacher"], {
  error: "userType must be one of: child, caregiver, teacher",
});

export const userTypeSchema = z
  .string({
    error: (iss) => (iss.input === undefined ? "userType is required" : "userType must be a string"),
  })
  .trim()
  .min(1, { error: "userType is required" })
  .toLowerCase()
  .pipe(UserType);
```